include/aspect/material_model/visco_plastic.h

```diff
@@ -221,16 +227,86 @@ namespace aspect
          * compositional field.
          */
         std::pair<double, double>
-        calculate_weakening ( const double strain_ii,
-                              const unsigned int j ) const;
+        calculate_plastic_weakening ( const double strain_ii,
+                                      const unsigned int j ) const;
+
+        /**
+         * A function that computes the strain weakened values
+         * of the diffusion and dislocation prefactors for a given
+         * compositional field.
+         */
+        double
+        calculate_viscous_weakening ( const double strain_ii,
+                                      const unsigned int j ) const;
 
+        /**
+         * Whether to use the accumulated strain to weaken
+         * plastic parameters cohesion and friction angle
+         * and viscous parameters diffusion and dislocation prefactor.
+         */
         bool use_strain_weakening;
+        /**
+         * Whether to use the accumulated plastic strain to weaken
+         * plastic parameters cohesion and friction angle.
+         * When both use_plastic_strain_weakening and
+         * use_viscous_strain_weakening are true, it is assumed
+         * the first 2 compositional fields are tracking the
+         * plastic and viscous strain respectively.
+         * TODO: add check for fields named plastic_strain and
+         * viscous_strain?
```

Comment:
Yes, I think it would be worth enforcing a naming scheme and having these values listed as the first and second compositional fields. The asserts could take the following form:

```
if (use_strain_weakening == true)
  {
    AssertThrow(this->introspection().compositional_index_for_name("plastic_strain") == 0,
                            ExcMessage("Material model ViscoPlastic only works if the first "
                                                   "compositional field is called plastic_strain."));
    AssertThrow(this->introspection().compositional_index_for_name("viscous_strain") == 1,
                           ExcMessage("Material model ViscoPlastic only works if the first "
                                                 "compositional field is called plastic_strain."));
  }
```

However, should similar functionality be include for when `use_strain_weakening == false` (first field could be `total_strain`) and for when we use the full strain tensor?